### PR TITLE
[tensorflow][build]  Added apt upgrades to Tf2.5-cpu,gpu,inf,train and updated Pillow to version 8.3.1

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

Adding APT Upgrade helps us consume security patches while rebuilding images and thus pass the sanity tests.

### DLC image/dockerfile

1. tensorflow/inference/docker/2.5/py3/Dockerfile.cpu
2. tensorflow/inference/docker/2.5/py3/cu112/Dockerfile.gpu
3. tensorflow/training/docker/2.5/py3/Dockerfile.cpu
4. tensorflow/training/docker/2.5/py3/cu112/Dockerfile.gpu

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/training/docker/2.5/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -136,7 +136,9 @@ RUN ln -s /etc/ssl/certs/*.* /usr/local/ssl/certs/
 RUN ln -s $(which ${PYTHON}) /usr/local/bin/python \
  && ln -s $(which ${PIP}) /usr/bin/pip
 
-RUN apt-get update && apt-get -y install protobuf-compiler
+RUN apt-get update \
+    && apt-get -y upgrade \
```
Comment (saimidu) [inline tensorflow/training/docker/2.5/py3/Dockerfile.cpu:140]: Please move this to the apt-get update/install `RUN` statement at the start of the dockerfile.

Also, please add a
```
 && rm -rf /var/lib/apt/lists/* \
 && apt-get clean
```
at the end of this `RUN` statement.

File: tensorflow/inference/docker/2.5/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -29,6 +29,7 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 # nginx + njs
 RUN apt-get update \
+ && apt-get -y upgrade \
```
Comment (saimidu) [inline tensorflow/inference/docker/2.5/py3/Dockerfile.cpu:32]: Please add this to the TF 1.15 dockerfiles as well.

File: dlc_developer_config.toml [hunk 3]
```diff
@@ -13,7 +13,7 @@ benchmark_mode = false
 
 [build]
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:14]: Please run SM test as well

**Review by arjkesh — DISMISSED** [review]

**Comment by arjkesh** [comment]
Please revert the config and we can get this merged in

**Review by arjkesh — APPROVED** [review]

**Pull request merged**